appIcons.js

```diff
@@ -231,14 +232,32 @@ const MyAppIcon = new Lang.Class({
         let event = Clutter.get_current_event();
         let modifiers = event ? event.get_state() : 0;
         let openNewWindow = modifiers & Clutter.ModifierType.CONTROL_MASK &&
-                            this.app.state == Shell.AppState.RUNNING ||
-                            button && button == 2;
+                            this.app.state == Shell.AppState.RUNNING;
         let focusedApp = tracker.focus_app;
 
         if (this.app.state == Shell.AppState.STOPPED || openNewWindow)
             this.animateLaunch();
 
-        if(button && button == 1 && this.app.state == Shell.AppState.RUNNING) {
+        // We check what type of click we have and if the modifier SHIFT is
+        // being used. We then define what _clickAction should be for this
+        // event.
+        let _clickAction = 0;
```

Comment:
Yeah, I put the prefix `_` since `clickAction` is already a global variable containing the different actions (line 32).

In this patch [e3a72723ca4750a75c689e894eaf956b60d3754d] I renamed it to `buttonAction`, since after all we first check what kind of button it is.